flax/training/train_state.py

```diff
@@ -15,10 +15,23 @@
 from typing import Any, Callable
 
 from flax import core
+from flax import linen as nn
 from flax import struct
+from flax import traverse_util
+
 import optax
 
 
+def _validate_params_axes(params_axes, params):
+  axis_names = nn.partitioning.get_axis_names(params_axes)
+  missing_params_axes = (
+      set(traverse_util.flatten_dict(params, sep='/')) -
+      set(traverse_util.flatten_dict(axis_names, sep='/')))
+  if missing_params_axes:
+    raise ValueError(
+        f'Missing axis names for parameters: {missing_params_axes}')
+
+
 class TrainState(struct.PyTreeNode):
```

Comment:
I might ask that we add this FP8 specific logic to a subclass of the general TrainState class? (it can be located in this file).